Sources/App/Models/Session/Session.swift

```diff
@@ -11,45 +13,64 @@ final class Session: Model {
   var userId: Identifier
   var token: String
   var actual: Bool
-  var timestamp: Int // NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP ??? FIX
   
   init(userId: Identifier,
        token: String,
-       actual: Bool = true,
-       timestamp: Int) {
+       actual: Bool = true) {
     self.userId = userId
     self.token = token
     self.actual = actual
-    self.timestamp = timestamp
   }
 
   // sourcery:inline:auto:Session.AutoModelGeneratable
   init(row: Row) throws {
     userId = try row.get(Keys.userId)
     token = try row.get(Keys.token)
     actual = try row.get(Keys.actual)
-    timestamp = try row.get(Keys.timestamp)
   }
 
   func makeRow() throws -> Row {
     var row = Row()
     try row.set(Keys.userId, userId)
     try row.set(Keys.token, token)
     try row.set(Keys.actual, actual)
-    try row.set(Keys.timestamp, timestamp)
     return row
   }
   // sourcery:end
 }
 
-extension Session {
+extension Session: JSONRepresentable {
   
+  func makeJSON() throws -> JSON {
+    var json = JSON()
+    try json.set(Keys.token, token)
+    try json.set(Keys.actual, actual)
+    return json
+  }
+}
+
+extension Session {
+
   static func find(by token: String) throws -> Session? {
     return try Session.makeQuery().filter(Keys.token, token).first()
   }
-  
+
   var user: User? {
     return try? parent(id: userId).get()!
   }
   
+  static func generateToken() throws -> String {
+    let random = try Crypto.Random.bytes(count: 16)
+    return random.base64Encoded.makeString()
+  }
+  
+  func updateToken() throws {
+    if
```

Comment:
I think this is the case for using `guard`

Reply:
You mean `guard` only for `let`s there? Because I also have a condition there `referenceDate < Date()`. So this code runs one time a month in fact and if it can't run it is not an error, except for `let`s initialization.

Reply:
```
guard
      let date = updatedAt,
      let referenceDate = Calendar.current.date(byAdding: .month, value: 1, to: date)
      else { throw Abort.serverError }
    if referenceDate < Date() {
      self.token = try Session.generateToken()
      try self.save()
    }
```
Like this?

Reply:
```
guard
  let date = updatedAt,
  let referenceDate = Calendar.current.date(byAdding: .month, value: 1, to: date),
  referenceDate < Date()
else { throw Abort.serverError }
token = try Session.generateToken()
try self.save()
```
I mean like this, u can add condition in `guard` as well. 
It's just I think, all places where it's only one `if` in function(without `else`), it's a perfect candidate for `guard`

Reply:
@dannydaddy3, `if referenceDate < Date()` you offer throw the server error? I guess it's wrong way.

Reply:
@distansu oh, i’m stupid) agree))